dll/win32/advapi32/reg/reg.c

```diff
@@ -1108,6 +1108,12 @@ RegCreateKeyExW(
 
     TRACE("RegCreateKeyExW() called\n");
 
+    if (lpSecurityAttributes && lpSecurityAttributes->nLength == 0)
+    {
+        lpSecurityAttributes = NULL;
```

Comment:
The caller is expected to properly set the length of the security attributes structure to `sizeof(SECURITY_ATTRIBUTES)` but if they don't do that then too bad, they're lazy. But the problem here now is that you are totally ignoring the security descriptor that was passed by the caller if they did not set the right structure size, thus the newly created key will get the default security descriptor instead of the one set up by the caller. Which is a hack.

Just remove the line below that returns `ERROR_INVALID_USER_BUFFER` -- it is completely pointless. `SECURITY_ATTRIBUTES` is a Win32 construct, the NT kernel only cares about a security descriptor and it is responsible to validate all of its contents in memory.

Reply:
Thanks, @GeoB99. That's an even better solution. Fewer lines of code and simpler. DONE!